### PR TITLE
Oauth2 in htmx

### PR DESCRIPTION
**Pull request by Firstyear**
Checklist

- [ x ] This pr contains no AI generated code
- [ x ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ x ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
Still needs me to fix some tests.

**Comment by Firstyear** [comment]
Fixes oauth2 and my bugs with healthcheck.

File: server/core/src/https/views/mod.rs [hunk 1]
```diff
@@ -29,38 +30,42 @@ struct UnrecoverableErrorView {
...
             post(login::view_login_mech_choose_post).get(|| async { Redirect::to("/ui") }),
         )
         .route(
-            "/api/login_backup_code",
+            "/login/backup_code",
             post(login::view_login_backupcode_post).get(|| async { Redirect::to("/ui") }),
         )
         .route(
-            "/api/login_totp",
+            "/login/totp",
```
Comment (yaleman) [inline server/core/src/https/views/mod.rs:64]: Ooh thanks for making it `/<action>/<type>`, I meant to pick that up at some point

Comment (Firstyear) [reply server/core/src/https/views/mod.rs:64]: Yeah, turns out this is good for cookie scoping :)

File: server/core/templates/oauth2_access_denied.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+(% extends "base.html" %)
+
+(% block title %)Oauth2 Access Denied(% endblock %)
```
Comment (yaleman) [inline server/core/templates/oauth2_access_denied.html:3]: We don't need to mention it's OAuth2 on this one
```suggestion
(% block title %)Access Denied(% endblock %)
```

File: server/core/templates/oauth2_access_denied.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+(% block head %)
+(% endblock %)
+
+(% block body %)
+	<h2>OAuth2 Access Denied</h2>
```
Comment (yaleman) [inline server/core/templates/oauth2_access_denied.html:9]:
```suggestion
	<h2>Access Denied</h2>
```

File: server/core/templates/oauth2_access_denied.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	<main id="main">
+		<p>Access to this service was denied as you do not have the requisite scopes.</p>
```
Comment (yaleman) [inline server/core/templates/oauth2_access_denied.html:11]: Requisite scopes etc isn't helpful for the user, just "denied" is enough in context.

```suggestion
```

File: server/core/templates/oauth2_consent_request.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
+(% extends "base.html" %)
+
+(% block title %)Oauth2 Access Denied(% endblock %)
```
Comment (yaleman) [inline server/core/templates/oauth2_consent_request.html:3]:
```suggestion
(% block title %)Consent Required(% endblock %)
```

File: server/core/templates/oauth2_consent_request.html [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+<main id="main" class="flex-shrink-0 form-signin">
+	<h2 class="h3 mb-3 fw-normal">Consent to Proceed to (( client_name ))</h2>
+	(% if pii_scopes.is_empty() %)
+		<div>
+			<p>This site will not have access to your personal information.</p>
+			<p>If this site requests personal information in the future we will check with you.</p>
+		</div>
+	(% else %)
+		<div>
+			<p>This site has requested to see the following personal information.</p>
```
Comment (yaleman) [inline server/core/templates/oauth2_consent_request.html:18]:
```suggestion
			<p>This site has requested access to the following personal information:</p>
```

File: server/daemon/src/main.rs [hunk 7]
```diff
@@ -1023,50 +1024,34 @@ async fn kanidm_main(
...
+                                            "Failed to parse {:?} into CA chain!\nError: {:?}",
+                                            ca_cert_path, e
                                         );
                                         return ExitCode::FAILURE;
                                     }
                                 };
-                            client.add_root_certificate(ca_cert_parsed)
+
+                            // Need at least 2 certs for the leaf + chain. We skip the leaf.
+                            for cert in ca_chain_parsed.into_iter().skip(1) {
```
Comment (yaleman) [inline server/daemon/src/main.rs:1051]: I sure hope they've got their end cert first in this case 😄

**Review by yaleman — CHANGES_REQUESTED** [review]
Minor changes but ... this is epic progress on the HTMX transition 🥳

File: server/daemon/src/main.rs (hunk 7 above)
Comment (Firstyear) [reply server/daemon/src/main.rs:1051]: Well if they don't it's not a valid cert chain :P

**Review by yaleman — APPROVED** [review]

**Pull request merged**